hri_safety_sense/src/VscProcess.cpp

```diff
@@ -15,207 +15,240 @@
-		}
+    if(msgPtr->EStopStatus > 0) {
+      RCLCPP_WARN(this->get_logger(), "Received ESTOP from the vehicle!!! 0x%x", msgPtr->EStopStatus);
+    }
 
-	} else {
-		ROS_WARN("RECEIVED HEARTBEAT WITH INVALID MESSAGE SIZE! Expected: 0x%x, Actual: 0x%x",
-				(unsigned int)sizeof(HeartbeatMsgType), recvMsg.msg.length);
-		retVal = 1;
-	}
+  } else {
+    RCLCPP_WARN(this->get_logger(), "RECEIVED HEARTBEAT WITH INVALID MESSAGE SIZE! Expected: 0x%x, Actual: 0x%x",
+      static_cast<unsigned int>(sizeof(HeartbeatMsgType)),
+      recvMsg.msg.meta.length);
+    retVal = 1;
+  }
 
-	return retVal;
+  return retVal;
 }
 
 void VscProcess::readFromVehicle()
 {
-	VscMsgType recvMsg;
-
-	/* Read all messages */
-	while (vsc_read_next_msg(vscInterface, &recvMsg) > 0) {
-		/* Read next Vsc Message */
-		switch (recvMsg.msg.msgType) {
-		case MSG_VSC_HEARTBEAT:
-			if(handleHeartbeatMsg(recvMsg) == 0) {
-				lastDataRx = ros::Time::now();
-			}
-
-			break;
-		case MSG_VSC_JOYSTICK:
-			if(joystickHandler->handleNewMsg(recvMsg) == 0) {
-				lastDataRx = ros::Time::now();
-			}
-
-			break;
-
-		case MSG_VSC_NMEA_STRING:
-//			handleGpsMsg(&recvMsg);
-
-			break;
-		case MSG_USER_FEEDBACK:
-//			handleFeedbackMsg(&recvMsg);
-
-			break;
-		default:
-			errorCounts.invalidRxMsgCount++;
-			ROS_ERROR("Receive Error.  Invalid MsgType (0x%02X)",recvMsg.msg.msgType);
-			break;
-		}
-	}
-
-	// Log warning when no data is received
-	ros::Duration noDataDuration = ros::Time::now() - lastDataRx;
-	if(noDataDuration > ros::Duration(.25)) {
-		ROS_WARN_THROTTLE(.5, "No Data Received in %i.%09i seconds", noDataDuration.sec, noDataDuration.nsec );
-	}
+  VscMsgType recvMsg;
+
+  /* Read all messages */
+  while (vsc_read_next_msg(vscInterface_, &recvMsg) > 0) {
+    /* Read next Vsc Message */
+    switch (recvMsg.msg.meta.msgType) {
+    case MSG_VSC_HEARTBEAT:
+      if(handleHeartbeatMsg(recvMsg) == 0) {
+        lastDataRx_ = this->now();
+      }
+
+      break;
+    case MSG_VSC_JOYSTICK:
+      if(joystickHandler_->handleNewMsg(recvMsg) == 0) {
+        lastDataRx_ = this->now();
+      }
+
+      break;
+
+    case MSG_VSC_NMEA_STRING:
+//      handleGpsMsg(&recvMsg);
+
+      break;
+    case MSG_USER_FEEDBACK:
+//      handleFeedbackMsg(&recvMsg);
+
+      break;
+    default:
+      errorCounts_.invalidRxMsgCount++;
+      RCLCPP_ERROR(this->get_logger(), "Receive Error.  Invalid MsgType (0x%02X)", recvMsg.msg.meta.msgType);
+      break;
+    }
+  }
+
+  // Log warning when no data is received
+  rclcpp::Duration noDataDuration = this->now() - lastDataRx_;
+  if(noDataDuration > rclcpp::Duration(0, 250000000)) {
+    // TODO(ros2/rclcpp#879) RCLCPP_THROTTLE_WARN() when released
+    THROTTLE(this->get_clock(), std::chrono::nanoseconds(500000000),
+      RCLCPP_WARN(this->get_logger(), "No Data Received in %g.%09i seconds",
+      noDataDuration.seconds(), noDataDuration.nanoseconds()));
+  }
 
 }
+}
 
```

Comment:
At the end here, you should add some rclcpp_component stuff.  Like [here](https://github.com/ros-drivers/velodyne/blob/8091dda8fe6fb1869963187ce593d11e5f80062d/velodyne_laserscan/src/velodyne_laserscan.cpp#L254).

Don't forget to add `rclcpp_components` to the package.xml and CMakeLists.txt

Reply:
695425a